### PR TITLE
feat: adding helm deployment support for AWS

### PR DESCRIPTION
**Pull request by deimosfr**
Implemented all charts for AWS and logic, not plugged yet

### CODE REVIEW

File: src/cloud_provider/helm.rs [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+use thread::spawn;
+
+#[derive(Clone)]
+pub enum HelmAction {
+    Deploy,
+    Destroy,
+    Skip,
+}
+
+#[derive(Copy, Clone)]
```
Comment (erebe) [inline src/cloud_provider/helm.rs:17]: Adding Debug trait can be useful

File: src/cloud_provider/helm.rs [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        HelmChartNamespaces::NginxIngress => "nginx-ingress",
+        HelmChartNamespaces::Qovery => "qovery",
+    }
+    .to_string()
+}
+
+pub trait HelmChart {
+    fn check_prerequisites(&self) -> Result<(), SimpleError> {
+        let chart = self.get_chart_info();
+        for file in chart.values_files.clone() {
```
Comment (erebe) [inline src/cloud_provider/helm.rs:81]: use into_iter, or iter()

File: src/cloud_provider/helm.rs (hunk 1 above)
Comment (deimosfr) [reply src/cloud_provider/helm.rs:17]: I'll see when doing the integration

File: src/cloud_provider/helm.rs (hunk 2 above)
Comment (deimosfr) [reply src/cloud_provider/helm.rs:81]: Using iter() won't let me escape from the lambda on the first error

File: src/cloud_provider/helm.rs [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        Ok(())
+    }
+
+    fn get_chart_info(&self) -> &ChartInfo;
+
+    fn namespace(&self) -> String {
+        get_chart_namespace(self.get_chart_info().namespace)
+    }
+
+    fn pre_exec(&self, _kubernetes_config: &Path, _envs: &Vec<(String, String)>) -> Result<(), SimpleError> {
```
Comment (erebe) [inline src/cloud_provider/helm.rs:104]: Change `&Vec<(String, String)>)` to `&[String, String)>]`

**Review by erebe — APPROVED** [review]

**Pull request merged**